src/components/Nav/Nav.tsx

```diff
@@ -196,6 +194,14 @@ export class Nav extends React.Component<INavProps, INavState> implements INav {
     ev.preventDefault();
     ev.stopPropagation();
   }
+
+  private _onNavLinkClicked(link: INavLink, ev: React.MouseEvent): void {
+    this._selectedKey = link.key;
+
+    if (link.onClick) {
+      link.onClick(ev);
```

Comment:
Ah! I see the correct fix. The anchor link and the button link depend on different onClicks. Modified the logic to reflect this again.